src/cluster_legacy.c

```diff
@@ -6501,7 +6501,7 @@ void clusterPromoteSelfToMaster(void) {
 
 int detectAndUpdateCachedNodeHealth(void) {
     dictIterator di;
-    dictInitSafeIterator(&di, server.cluster->nodes);
+    dictInitIterator(&di, server.cluster->nodes);
```

Comment:
there is another misuse: https://github.com/valkey-io/valkey/blob/unstable/src/cluster_legacy.c#L4895